_transcripts/031-rust-1.46-1.47.md

```diff
@@ -0,0 +1,894 @@
+they want to run instead. What CFG does, the Control Flow Guard, is basically
+inject this little check, just before the call to F, to check that F is still a
+valid target, that it doesn't point into some, like, bad memory or something.
+And if it does, then it just terminates the process. There's a little bit of
+sort of coordination that's required here between the compiler and the Windows
+kernel at runtime that keeps track of, like, what are valid jump targets, like
+what are valid functions to call, and such. So it basically prevents some
+attacks that are bad when they happen. I don't know how effective it is at
+preventing real attacks, like in the wild, but it certainly seems like something
+that that could be a pretty good idea, to just—
+
+__Ben__: I'm sure it's based on things that have been observed in the wild. I
+know that, like, it might not be as useful for Rust in particular, in terms of,
+like, if you're writing `unsafe` code in Rust, I'm sure it is, but it's not
+actually turned on by default in `rustc`, because I think the idea is that
+almost all Rust code actually doesn't need this. Obviously, there are plenty of
+things in Rust, that Rust does turn on by default. And so you have, like,
+typical hardening things, like position independent code and all the other— we
+have stack guards and stack canaries and things that kind of like, even though
+normally you wouldn't encounter these. Sometimes it's just good to have for
+additional security. But I think in this case, it was just decided that the
+performance tradeoff to actual risk was not good enough to turn it on. But
+obviously, you can turn it on if you so elect to, with this new flag to `rustc`.
+
+__Jon__: Yeah, that sounds about right.
+
+There were a bunch of smaller library changes in 1.47 as well. And I think
+before we go to some of the specifics, do you want to talk a little bit about
+the `const` changes here?
+
+__Ben__: Yeah. So as we mentioned before in the previous release, `const` now
+has the capability of doing a bunch of `if`s and `loop`s. And so there are
+plenty of newly-const methods. In this case, although the scope is pretty
+narrow, the biggest group are probably: for all integers, there are various
+operations. There's `checked_add`, `checked_sub`-traction,
+`checked_mul`-tiplication, and by "checked" we mean that in this case, if it
+would overflow, it will provide an error. Also `saturating_add`,
+`saturating_sub`, etc. In this case, if it would overflow, it will— or under
+flow as well, I'm pretty sure, I guess in subtraction. I guess you can also add
+a negative in the signed case.
+
+__Jon__: Yeah, it's—
+
+__Ben__: But yeah, in either case, instead of overflowing, they will just kind
+of clamp themselves to the highest value. And so, if you try to add two `MAX`
+ints, you just get `MAX` int back. Or try to subtract zero or, you know, like a
+`MAX` int from zero, you just get zero back. And so different people want
```

Comment:
`MAXINT`? `MAX_INT`? I'm not sure how to render this, so I guessed at something vaguely Rust-y (audio 35:22).

Reply:
I think the way you did it is probably just as understandable as the alternatives, and looks slightly nicer :sweat_smile: